### ISSUE TITLE
Mac, ignoring shoes-core-4.0.0.pre5 because its extensions are not built...

### ISSUE THREAD

**Issue by maxiedaniels** [shoes/shoes4#1239]
I'm trying to get shoes4 working on my Mac, system is El Capitan. First I've tried just 'gem install shoes --pre', which is the least broken for me. When I run shoes <myfile>.rb, I get "Ignoring shoes-core-4.0.0.pre5 because its extensions are not built.  Try: gem pristine shoes-core --version 4.0.0.pre5", and then a whole lot of errors about undefined methods.. ex. 'last_line_width', "contents_alignment'... etc. The app then displays, but the alignments are completely wrong.

I've tried the gem pristine command.. no dice. Nothing changes.

I've also tried the git clone + bundler method, but that seems to be completely broken for me. I'll start with a clean setup and do that method, but when I go to run shoes, it says a bunch of gems have extensions not built (ffi, gem-wrappers, nokogiri, unf_ext) and then it says /usr/local/bin/shoes: shoes4: command not found.

Any ideas? I have rvm, with ruby-head and jruby installed and ruby-head selected. I actually had to install rvm because El Capitan has the SIP feature which screws up gems in the system folder.

**Comment by PragTob** [comment]
Hi there,

shoes does not work with ruby-head. Please try Jruby, preferable JRuby 9.1+ :)

**Comment by maxiedaniels** [comment]
Ah - that would explain a lot. Thank you! I assumed it was running jRuby from within Shoes and that I didn't need to select jRuby as my current Ruby.

**Comment by PragTob** [comment]
is this your error: #1217 ?

If so, then installing from master or a JRuby prior to 9.0.5.0 should help (they introduced a breaking change). And I should make a new gem release :)

Cheers,
Tobi

**Comment by maxiedaniels** [comment]
Yup that's the error. But I'm trying to install from the master... when I run shoes now, I get:

`Ignoring executable-hooks-1.3.2 because its extensions are not built.  Try: gem pristine executable-hooks --version 1.3.2
Ignoring gem-wrappers-1.2.7 because its extensions are not built.  Try: gem pristine gem-wrappers --version 1.2.7
Ignoring jruby-launcher-1.1.1-java because its extensions are not built.  Try: gem pristine jruby-launcher --version 1.1.1
/usr/local/lib/ruby/site_ruby/2.3.0/rubygems/dependency.rb:308:in to_specs: Could not find 'shoes-core' (>= 0) among 72 total gem(s) (Gem::MissingSpecError)
Checked in 'GEM_PATH=/Users/maxfarrar/.rvm/gems/jruby-9.0.5.0:/Users/maxfarrar/.rvm/gems/jruby-9.0.5.0@global', execute gem env for more information`

Another weird thing - when I cd into the shoes4 (git cloned directory), I get:
`Unknown ruby string (do not know how to handle): jruby-9.1.1.0.
jruby-9.1.1.0 is not installed.`

Again, that's just from typing `cd shoes4.` It does go into the directory, but I've never seen cd cause a command other than cd to run...

**Comment by PragTob** [comment]
@maxiedaniels you have rvm installed, rvm hooks into cd and looks for `.ruby-version` which defines which ruby version to use and then tries to switch to it. Try `rvm get head` and then `rvm install jruby-9.1.1.0` that should install it.

Otherwise, installing from master doesn't build the master (unless you build the gem and install it) but you gotta rather use the executable in the repo referenced by path - it's at `bin shoes` :)

**Comment by maxiedaniels** [comment]
Interesting... Thank you for the info. So if bundle install isn't building the master, how do I build it? I don't want to use the gem since it has the bug. I've installed jruby-9.1.1.0 but no difference except that error doesn't show.

**Comment by jasonrclark** [comment]
Hey @maxiedaniels. Been a long time on this, but we've released a new version of the `shoes` gem(s) which address the `line_bounds` error you were hitting. I've been running against JRuby 9.1.6.0 with that for a bit and things look tidy on that front.

If I don't hear anything from you, I'll close this out as the specific issues appear to have been addressed, but would love confirmation from you if you've got time!
